packages/searchkit/src/highlightUtils.ts

```diff
@@ -6,6 +6,21 @@ export function highlightTerm(value: string, query: string): string {
   return value.replace(regex, (match) => `<em>${match}</em>`)
 }
 
+export function isAllowableHighlightField(
+  fieldKey: string,
+  highlightFields: string[]
+) {
+  return highlightFields.findIndex((highlightField) => {
+    if (highlightField.indexOf('*') < 0) {
+      return highlightField === fieldKey;
+    }
+
+    const safeHighlightField = highlightField.replace(/[.+?^$|\{\}\(\)\[\]\\]/g, '\\$&');
```

Comment:
I made the assumption here that we should probably escape regex special characters from field names defined in `highlight_attributes`. I don't know the full whitelist of allowable characters - if this is excessive and unnecessary I can drop it to speed things up a bit.